packages/core/saved-objects/core-saved-objects-server/docs/model_versions.md

```diff
@@ -871,6 +876,148 @@ const myType: SavedObjectsType = {
 };
 ```
 
+## Testing model versions
+
+Model versions definitions are more structured than the legacy migration functions, which makes them harder
+to test without the proper tooling. This is why a set of testing tools and utilities are exposed
+from the `@kbn/core-test-helpers-model-versions` package, to help properly test the logic associated
+with model version and their associated transformations.
+
+### Tooling for unit tests
+
+For unit tests, the package exposes utilities to easily test the impact of transforming documents 
+from a model version to another one, either upward or backward.
+
+#### Model version test migrator
+
+The `createModelVersionTestMigrator` helper allows to create a test migrator that can be used to 
+test model version changes between versions, by transforming documents the same way the migration
+algorithm would during an upgrade.
+
+**Example:**
+
+```ts
+import { 
+  createModelVersionTestMigrator, 
+  type ModelVersionTestMigrator 
+} from '@kbn/core-test-helpers-model-versions';
+
+const mySoTypeDefinition = someSoType();
+
+describe('mySoTypeDefinition model version transformations', () => {
+  let migrator: ModelVersionTestMigrator;
+  
+  beforeEach(() => {
+    migrator = createModelVersionTestMigrator({ type: mySoTypeDefinition });
+  });
+  
+  describe('Model version 2', () => {
+    it('properly backfill the expected fields when converting from v1 to v2', () => {
+      const obj = createSomeSavedObject();
+
+      const migrated = migrator.migrate({
+        document: obj,
+        fromVersion: 1,
+        toVersion: 2,
+      });
+
+      expect(migrated.properties).toEqual(myExpectedProperties);
+    });
+
+    it('properly removes the expected fields when converting from v2 to v1', () => {
+      const obj = createSomeSavedObject();
+
+      const migrated = migrator.migrate({
+        document: obj,
+        fromVersion: 2,
+        toVersion: 1,
+      });
+
+      expect(migrated.properties).toEqual(myExpectedProperties);
+    });
+  });
```

Comment:
For best practices, should we recommend having a global `myExpectedProperties` for each version so that we can test symmetric migrations (when possible)